pkg/client/informers/informers_generated/internalversion/factory.go

```diff
@@ -44,36 +44,63 @@ import (
 	storage "k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/storage"
 )
 
+type ResyncConfiguration map[v1.Object]time.Duration
+type SharedInformerOption func(*sharedInformerFactory) *sharedInformerFactory
+
 type sharedInformerFactory struct {
 	client           internalclientset.Interface
 	namespace        string
 	tweakListOptions internalinterfaces.TweakListOptionsFunc
 	lock             sync.Mutex
 	defaultResync    time.Duration
+	customResync     map[reflect.Type]time.Duration
 
 	informers map[reflect.Type]cache.SharedIndexInformer
 	// startedInformers is used for tracking which informers have been started.
 	// This allows Start() to be called multiple times safely.
 	startedInformers map[reflect.Type]bool
 }
 
-// NewSharedInformerFactory constructs a new instance of sharedInformerFactory
-func NewSharedInformerFactory(client internalclientset.Interface, defaultResync time.Duration) SharedInformerFactory {
-	return NewFilteredSharedInformerFactory(client, defaultResync, v1.NamespaceAll, nil)
+// Sets a custom resync period for the specified informer types.
+func WithCustomResyncConfig(resyncConfig ResyncConfiguration) SharedInformerOption {
+	return func(factory *sharedInformerFactory) *sharedInformerFactory {
+		for k, v := range resyncConfig {
+			factory.customResync[reflect.TypeOf(k)] = v
+		}
+		return factory
+	}
+}
+
+// Sets a custom filter on all listers of the configured SharedInformerFactory.
+func WithTweakListOptions(tweakListOptions internalinterfaces.TweakListOptionsFunc) SharedInformerOption {
+	return func(factory *sharedInformerFactory) *sharedInformerFactory {
+		factory.tweakListOptions = tweakListOptions
+		return factory
+	}
+}
+
+// NewSharedInformerFactory constructs a new instance of sharedInformerFactory with additional options.
+func NewSharedInformerFactory(client internalclientset.Interface, defaultResync time.Duration, options ...SharedInformerOption) SharedInformerFactory {
+	return NewFilteredSharedInformerFactory(client, defaultResync, v1.NamespaceAll, options...)
 }
 
-// NewFilteredSharedInformerFactory constructs a new instance of sharedInformerFactory.
-// Listers obtained via this SharedInformerFactory will be subject to the same filters
-// as specified here.
-func NewFilteredSharedInformerFactory(client internalclientset.Interface, defaultResync time.Duration, namespace string, tweakListOptions internalinterfaces.TweakListOptionsFunc) SharedInformerFactory {
-	return &sharedInformerFactory{
+// NewFilteredSharedInformerFactory constructs a new instance of sharedInformerFactory contained to the specified namespace with additional options.
+func NewFilteredSharedInformerFactory(client internalclientset.Interface, defaultResync time.Duration, namespace string, options ...SharedInformerOption) SharedInformerFactory {
```

Comment:
we have to leave the old func and reimplement it with `NewInformerFactoryWithOptions`.

Reply:
Disregard my previous comment. it's compatible. But still, I would not add the options here. Instead I would even deprecate it in favor of the generic other variant with options.

Reply:
As `NewFilteredSharedInformerFactory` is not used anywhere I deleted it and introdued `NewSharedInformerFactoryWithOptions`.

Reply:
It's used by 3rdparties. Would leave it and just deprecate it.

Reply:
Done.